relayer/client-tx.go

```diff
@@ -191,3 +203,38 @@ func (c *Chain) UpgradeClients(dst *Chain) error {
 
 	return nil
 }
+
+// FindMatchingClient will determine if there exists a client with identical client and consensus states
+// to the client which would have been created.
+func FindMatchingClient(c *Chain, msg sdk.Msg, consensusStateHeight exported.Height) (string, bool) {
+	createClientMsg, ok := msg.(*clienttypes.MsgCreateClient)
+	if !ok {
+		return "", false
+	}
+
+	// TODO: add appropriate offset and limits, along with retries
+	clientsResp, err := c.QueryClients(0, 1000)
+	if err != nil {
+		return "", false
+	}
+
+	for _, identifiedClientState := range clientsResp.ClientStates {
+		// check if the client states match
+		if proto.Equal(createClientMsg.ClientState, identifiedClientState.ClientState) {
+
+			// check if consensus state exists
+			consensusStateResp, err := clientutils.QueryConsensusStateABCI(c.CLIContext(0), identifiedClientState.ClientId, consensusStateHeight)
+			if err != nil {
+				// consensus state does not exist, try next client
+				continue
+			}
+
+			if proto.Equal(consensusStateResp.ConsensusState, createClientMsg.ConsensusState) {
+				// matching client found
+				return identifiedClientState.ClientId, true
+			}
```

Comment:
Doing this makes it unlikely that the consensus state actually matches. We don't want to rely on the `ConsensusState` in the `CreateClientMsg` message. 

Why can't we just take the latest consensus state for that client state, we can then just check if that consensus state matches the block at the specified height on chain `c`. If it does, then we can be sure it's a client of the same chain.

Reply:
perfect, exactly the feedback I was hoping for. That makes a lot of sense and should actually make client reuse viable

Reply:
Ahh it looks like `c` is the chain that you are sending the `CreateClientMsg` to. You want to query the header from the other chain, since that is the chain you are creating the client for. Will require a refactor, but still possible.